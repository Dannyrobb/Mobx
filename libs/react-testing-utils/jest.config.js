module.exports = {
  displayName: 'react-testing-utils',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/react-testing-utils',
};
