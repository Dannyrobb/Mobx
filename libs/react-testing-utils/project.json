{
  "root": "libs/react-testing-utils",
  "sourceRoot": "libs/react-testing-utils/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nrwl/web:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/react-testing-utils",
        "tsConfig": "libs/react-testing-utils/tsconfig.lib.json",
        "project": "libs/react-testing-utils/package.json",
        "entryFile": "libs/react-testing-utils/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": "@nrwl/react/plugins/bundle-rollup",
        "compiler": "babel",
        "assets": [
          {
            "glob": "libs/react-testing-utils/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/react-testing-utils/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/react-testing-utils"],
      "options": {
        "jestConfig": "libs/react-testing-utils/jest.config.js",
        "passWithNoTests": true
      }
    }
  }
}
