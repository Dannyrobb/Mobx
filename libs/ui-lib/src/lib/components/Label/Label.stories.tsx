import React from 'react';

import { Story, Meta } from '@storybook/react';

import { Label, LabelProps } from './Label';

export default {
  title: 'UI Lib / Label',
  component: Label,
  argTypes: {},
} as Meta;

const Template: Story<LabelProps> = (args) => (
  <>
    <Label {...args} />
  </>
);

export const base = Template.bind({});
base.args = {
  label: 'This Label',
};
