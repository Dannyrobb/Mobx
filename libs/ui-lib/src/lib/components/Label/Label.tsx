import * as React from 'react';

import { theme } from '@cellxpert/theme';
import styled from '@emotion/styled';

type Variant = 'small' | 'big';

export interface LabelProps {
  label: string;
  /**
   * In case other flavors are needed.
   * @default small
   */
  variant?: Variant;
}

const StyledLabel = styled('label')(() => ({
  fontFamily: theme.typography.fontFamily[0],
  fontSize: 14,
  color: theme.palette.text.main,
  lineHeight: '20px',
  fontWeight: 600,
}));

export const Label: React.FunctionComponent<LabelProps> = ({ label, variant = 'small', ...rest }) => {
  return <StyledLabel {...{ ...rest, variant }}>{label}</StyledLabel>;
};

export default Label;
