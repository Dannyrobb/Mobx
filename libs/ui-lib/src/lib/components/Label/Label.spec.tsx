import { render } from '@cellxpert/react-testing-utils';

import { Label } from './Label';

describe('Label', () => {
  it('should render successfully', () => {
    const { baseElement } = render(<Label label="Label" />);

    expect(baseElement).toBeTruthy();
  });
  it('snould not have a label attribute', () => {
    const { baseElement, getByText } = render(<Label label="Label" />);

    expect(baseElement).toMatchSnapshot();
    // expect(getByText('Label')).not.toHaveAttribute('label');
  });
});
