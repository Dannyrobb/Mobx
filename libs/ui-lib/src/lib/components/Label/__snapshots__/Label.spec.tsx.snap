// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label snould not have a label attribute 1`] = `
<body>
  <div>
    <label
      class="css-1l34zne-StyledLabel e18hlgk20"
    >
      Label
    </label>
  </div>
</body>
`;
